packages/node/src/node.ts

```diff
@@ -101,58 +111,56 @@ export class Node {
             "Reached OP_SIGN middleware without generated commitment."
           );
         }
-        context.signature = {
-          v: -1,
-          r: HashZero,
-          s: HashZero
-        };
-        // TODO: Replace the above hack with the below
-        // context.signature = this.signer.signDigest(
-        //   context.commitment.hashToSign()
-        // );
+
+        context.signature = this.signer.signDigest(
+          context.commitment.hashToSign()
+        );
+
         next();
       }
     );
-  }
 
-  private registerIoMiddleware() {
-    this.instructionExecutor.register(
+    instructionExecutor.register(
       Opcode.IO_SEND,
       async (message: ProtocolMessage, next: Function, context: Context) => {
-        await this.messagingService.send(context.outbox[0].toAddress, {
-          from: this.address,
-          event: NODE_EVENTS.PROTOCOL_MESSAGE_EVENT,
-          data: context.outbox[0]
-        });
+        const from = getAddress(this.address);
+        const to = getAddress(context.outbox[0].toAddress);
+        const data = context.outbox[0];
```

Comment:
It's also kind of confusing that the thing being sent to the other party isn't the `message`, but rather `context.outbox[0]` which isn't obvious as to what it actually contains.

Reply:
The problem is an overlapping of terminology. The first message refers to the object passed between opcodes in an instruction flow in the middleware-architecture based instruction executor. The second message refers to the NodeMessages received from your counterparty.